packages/ember-template-compiler/lib/plugins/safe-integers-bugfix.ts

```diff
@@ -0,0 +1,86 @@
+import { AST, ASTPlugin, ASTPluginEnvironment, Syntax } from '@glimmer/syntax';
+import { MustacheStatement, NumberLiteral } from '@glimmer/syntax/dist/types/lib/types/nodes';
+
+/**
+ @module ember
+*/
+
+/**
+  A Glimmer2 AST transformation that replaces all instances of
+
+  ```handlebars
+ {{123456789}}
+  ```
+
+  to
+
+  ```handlebars
+ {{-i "123456789"}}
+  ```
+
+  as well as other integer number literals in sexp arguments, etc.
+
+  The version of Glimmer VM we are using has a bug that encodes
+  certain integers incorrectly. This forces them into strings and
+  use `{{-i}}` (which is a wrapper around `parseInt`) to decode
+  them manually as a workaround.
+
+  This should be removed when the Glimmer VM bug is fixed.
+
+  @private
+  @class SafeIntegersBugfix
+*/
+
+export default function safeIntegersBugfix(env: ASTPluginEnvironment): ASTPlugin {
+  let { builders: b } = env.syntax;
+
+  return {
+    name: 'safe-integers-bugfix',
+
+    visitor: {
+      MustacheStatement(node: AST.MustacheStatement): AST.MustacheStatement | undefined {
+        if (!requiresWorkaround(node)) {
+          return;
+        }
+
+        return b.mustache(
+          '-i',
+          [b.string(String(node.path.value))],
+          undefined,
+          !node.escaped,
+          node.loc
+        );
+      },
+
+      NumberLiteral(node: AST.NumberLiteral): AST.SubExpression | undefined {
+        if (!requiresWorkaround(node)) {
+          return;
+        }
+
+        return b.sexpr(
+          '-i',
+          [b.string(String(node.value))],
+          undefined,
+          node.loc
+        );
+      },
+    },
+  };
+}
+
+function requiresWorkaround(node: AST.MustacheStatement): node is MustacheStatement & { path: NumberLiteral };
+function requiresWorkaround(node: AST.NumberLiteral): boolean;
+function requiresWorkaround(node: AST.MustacheStatement | AST.NumberLiteral): boolean {
+  if (node.type === 'MustacheStatement' && node.path.type === 'NumberLiteral') {
+    return requiresWorkaround(node.path);
+  } else if (node.type === 'NumberLiteral') {
+    // TODO: narrow this down?
+    return isInteger(node);
+  } else {
+    return false;
+  }
+}
+
+function isInteger(node: AST.NumberLiteral): boolean {
+  return Number.isInteger(node.value);
```

Comment:
`isInteger` is not supported by `IE` although I'm guessing this code is only going to execute in `node.js`.

Reply:
That’s a good point since we support running the template compiler in the browser